mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -311,24 +315,21 @@ fn get_url_for_get_id(url_host: &str, device_id: &str) -> String {
     url_get_id
 }
 
-async fn get_jwt_token(client: &Client) -> Result<SmartRestJwtResponse, SMCumulocityMapperError> {
-    let mut subscriber = client.subscribe(Topic::new("c8y/s/dat")?.filter()).await?;
-
+async fn get_jwt_token(
+    client: &Client,
+    receiver: &async_channel::Receiver<String>,
+) -> Result<SmartRestJwtResponse, SMCumulocityMapperError> {
```

Comment:
This can be improved passing a single argument: `&self`.

* Indeed on the call site, both the `client` and the `receiver` are taken from `self`.
* Furthermore, this has no chance to work unrelated `client` and `receiver` that have not been prepared by `self`.

Reply:
done